pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -432,8 +433,30 @@ func (f *ring0Factory) Printer(mapping *meta.RESTMapping, options printers.Print
 }
 
 func (f *ring0Factory) Pauser(info *resource.Info) ([]byte, error) {
+	_, isUnstructured := info.Object.(runtime.Unstructured)
+	if isUnstructured {
+		info.Object = info.AsVersioned()
```

Comment:
@smarterclayton is overwriting info.Object like this expected?

Reply:
I guess this is fine, since we are already doing that:
https://github.com/kubernetes/kubernetes/blame/4e2f5e2212b05a305435ef96f4b49dc0932e1264/pkg/kubectl/cmd/set/set_env.go#L325
https://github.com/kubernetes/kubernetes/blame/4e2f5e2212b05a305435ef96f4b49dc0932e1264/pkg/kubectl/cmd/set/set_image.go#L193
https://github.com/kubernetes/kubernetes/blame/4e2f5e2212b05a305435ef96f4b49dc0932e1264/pkg/kubectl/cmd/set/set_resources.go#L196
https://github.com/kubernetes/kubernetes/blame/4e2f5e2212b05a305435ef96f4b49dc0932e1264/pkg/kubectl/cmd/set/set_serviceaccount.go#L155

Reply:
in a particular set command seems different than in a factory method

Reply:
I see 2 options here:
- Move the factory into its corresponding cmd's code,
`func Pauser()` and `func Resumer()` are used only by `kubectl rollout pause` and `kubectl rollout resume` respectively.
- Find another place to convert `Unstructured` to `Versioned`, probably in `kubectl rollout pause` and `kubectl rollout resume` code.

Reply:
I'm leaning to option 1, since it can make the factory code thin by moving some unnecessary functions out.

Reply:
I'd leave it in the factory, but avoid overwriting info.Object... just do the conversion local to the function

Reply:
> I'd leave it in the factory, but avoid overwriting info.Object... just do the conversion local to the function

I'm not sure what's the strong reason we removed `VersionObject`,but l'd like to see what it looks like with @liggitt 's suggestion.